### PR TITLE
[Stream Support]: Added streaming support from GCS for file resource.

### PR DESCRIPTION
**Pull request by anshul-patel-infostretch**
- Updated `iter_content()` method to allow streaming from GCS.
- Updated the documentation.

## Verification

* make test, lint and integration works.

## Testing

* Uploaded a sample with size > 5 MB.
* Streamed via `iter_content()` to local temporary file.
* Verified the consistency via md5 checksum.

### CODE REVIEW

File: docs/downloading.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ file.download("/tmp/file.csv")
 
 ## Download streaming chunks
 
-Download a file in chunks of bytes, for example to stream out while downloading.
+Download a file in chunks of bytes, for example to stream out while downloading from Google Storage.
```
Comment (antoncohen) [inline docs/downloading.md]: I don't think this needs to mention "Google Storage", it is an implementation detail.

File: crux/models/file.py [hunk 2]
```diff
@@ -216,9 +218,20 @@ def iter_content(self, chunk_size=DEFAULT_CHUNK_SIZE):
         if not valid_chunk_size(chunk_size):
             raise ValueError("chunk_size should be multiple of 256 KiB")
 
-        data = self.connection.api_call(
-            "GET", ["resources", self.id, "content"], headers=headers, stream=True
-        )
+        # If we must use only Crux domains, download via the API.
+        if only_use_crux_domains:
```
Comment (antoncohen) [inline crux/models/file.py]: This needs to have the [same logic](https://github.com/cruxinformatics/crux-python/blob/2308a4ebb847572ace30ad731a35508d32b7cbd9/crux/models/file.py#L236) as the regular file download, where it gets the value from `crux_config.only_use_crux_domains` if it isn't explicitly passed to the method.

File: docs/downloading.md [hunk 3]
```diff
@@ -50,6 +50,25 @@ finally:
     stream.close()
 ```
 
+Download a file in chunks of bytes, for example to stream out while downloading from Crux domain.
```
Comment (antoncohen) [inline docs/downloading.md]: I wouldn't include this. We don't really want people restricting the downloads to Crux domains if they don't need to.

File: docs/downloading.md (hunk 1 above)
Comment (anshul-patel-infostretch) [reply docs/downloading.md]: Thanks and Agreed, updated the code.

File: crux/models/file.py (hunk 2 above)
Comment (anshul-patel-infostretch) [reply crux/models/file.py]: My bad, it should have been taken care in first place, will take care in future.

File: docs/downloading.md (hunk 3 above)
Comment (anshul-patel-infostretch) [reply docs/downloading.md]: Thanks and Agreed, updated the code.

**Pull request merged**